pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,272 @@ type StorageVolumeList struct {
+	OnlinePassive RecoveryTimeObjectives = "OnlinePassive"
+)
+
+type ProvisioningPolicy string
+
+const (
+	// ProvisioningPolicyThick indicates that the provisioning policy is thick
+	ProvisioningPolicyThick ProvisioningPolicy = "Thick"
+
+	// ProvisioningPolicyThin indicates that the provisioning policy is thin
+	ProvisioningPolicyThin ProvisioningPolicy = "Thin"
+)
+
+type CompressionAlgorithm string
+
+const(
+	HuffmanCompression CompressionAlgorithm = "huffman"
+)
+
+type DataSecurityCapabilities struct {
+	MediaEncryption *bool `json:"MediaEncryption,omitempty"`
+}
+
+type IOConnectivityCapabilities struct {
+	AccessProtocols string `json:"accessProtocols,omitempty"`
+}
+
+type IOPerformanceCapabilities struct {
+	AverageIOOperationLatencyMicroseconds string `json:"averageIOOperationLatencyMicroseconds,omitempty"`
+}
+
+type DataProtectionCapabilities struct {
+	// TODO: Decide the Fields
+}
+
+// Affinity is a group of affinity scheduling rules
+type Affinity struct {
+
+	// Describes volume affinity scheduling rules
+	// +optional
+	VolumeAffinity []VolumeAffinityTerm `json:"volumeAffinity,omitempty"`
+
+	// Describes volume anti-affinity scheduling rules
+	// +optional
+	VolumeAntiAffinity []VolumeAffinityTerm `json:"volumeAntiAffinity,omitempty"`
+
+	// Describes cohort affinity scheduling rules
+	// +optional
+	CohortAffinity []metav1.LabelSelector `json:"cohortAffinity,omitempty"`
+
+	// Describes cohort anti-affinity scheduling rules
+	// +optional
+	CohortAntiAffinity []metav1.LabelSelector `json:"cohortAntiAffinity,omitempty"`
+
+	// TopologySpreadConstraint specifies how to spread matching volumes among the given topology.
+	// +optional
+	TopologySpreadConstraint []v1.TopologySpreadConstraint `json:"topologySpreadConstraint,omitempty"`
+}
+
+type VolumeAffinityTerm struct {
+	TopologyKey string `json:"topologyKey,omitempty"`
+
+	LabelSelector *metav1.LabelSelector `json:"labelSelector,omitempty"`
+}
+
+// StorageVolumeStatus defines the observed state of StorageVolume
+// +k8s:openapi-gen=true
+type StorageVolumeStatus struct {
+	// Phase defines the state of the volume
+	Phase StorageVolumePhase `json:"phase"`
+
+	// Capacity stores total, used and available size
+	// +nullable
+	Capacity Capacity `json:"capacity,omitempty"`
+
+	// NVMESpec defines the nvme spec i.e nql, targetIP and status
+	// +nullable
+	NVMESpec []NVMESpec `json:"NVMESpec,omitempty"`
+}
+
+// Capacity defines total, used and available size
+type Capacity struct {
+	// Total represents total size of volume
+	Total resource.Quantity `json:"total,omitempty"`
+
+	// Used represents used size of the volume
+	Used resource.Quantity `json:"used,omitempty"`
+
+	// Available represents available size of the volume
+	Available resource.Quantity `json:"available,omitempty"`
+
+	// Error denotes the error occurred during provisioning a volume.
+	// Error field should only be set when State becomes Failed.
+	Error *StorageVolumeError `json:"error,omitempty"`
```

Comment:
why error is inside capacity?

Reply:
Ahh, by mistake I have placed it over here. It should be inside status.